server/service/src/stocktake/insert/validate.rs

```diff
@@ -12,12 +12,18 @@ pub fn validate(
     store_id: &str,
     stocktake: &InsertStocktake,
 ) -> Result<(), InsertStocktakeError> {
-    if !check_stocktake_does_not_exist(connection, &stocktake.id)? {
+    if stocktake.is_initial_stocktake == true {
+        if !check_initial_stocktake_does_not_exist(connection, store_id)? {
+            return Err(InsertStocktakeError::InitialStocktakeAlreadyExists);
+        }
+    }
```

Comment:
You can use 'not/bang' operators for booleans and the two conditions can be combined into one~